### PR TITLE
add a script to run the baremetal operator in demo mode

### PR DESCRIPTION
**Pull request by dhellmann**
It's easy to misconfigure the demo-mode server, so let's automate it
to make it simpler.

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/493/

File: demo/launch_demo_bmo.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# Run the baremetal operator locally in "demo" mode.
+#
+# This script assumes that you have ~/.kube/config set up, which may
+# mean copying ocp/auth/kubeconfig after running the scripts to build
+# a dev environment.
+
+set -xe
+
+eval $(go env)
```
Comment (hardys) [inline demo/launch_demo_bmo.sh]: As well as eval-ing the go env, we need to set the KUBECONFIG for `oc` below, either via common.sh:

```
SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
source ${SCRIPTDIR}/../common.sh
```

Or via a similar method that sets only KUBECONFIG to `$SCRIPTDIR/../ocp/auth/kubeconfig`?

File: demo/launch_demo_bmo.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    oc delete deployment metalkube-baremetal-operator
+fi
+
+cd $GOPATH/src/github.com/metalkube/baremetal-operator
+
+oc apply -f deploy/crds/demo-hosts.yaml
+
+export OPERATOR_NAME=baremetal-operator
+
+operator-sdk up local \
```
Comment (hardys) [inline demo/launch_demo_bmo.sh]: We either need to add $GOPATH/bin to the path or use an absolute path here

**Comment by hardys** [comment]
Looks good, thanks - I tested with the following changes as per comments:

```
[shardy@dell-r630-007 dev-scripts]$ git diff
diff --git a/demo/launch_demo_bmo.sh b/demo/launch_demo_bmo.sh
index 0343371..462ac82 100755
--- a/demo/launch_demo_bmo.sh
+++ b/demo/launch_demo_bmo.sh
@@ -8,6 +8,8 @@
 
 set -xe
 
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source ${SCRIPTDIR}/../common.sh
 eval $(go env)
 
 # Set the project
@@ -26,6 +28,6 @@ oc apply -f deploy/crds/demo-hosts.yaml
 
 export OPERATOR_NAME=baremetal-operator
 
-operator-sdk up local \
+${GOPATH}/bin/operator-sdk up local \
     --namespace=openshift-machine-api \
        --operator-flags="-dev -demo-mode"
```

**Comment by dhellmann** [comment]
@hardys thanks for testing this out. I've made the changes you suggested.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/506/

**Pull request merged**